soc/intel/intel_adsp/tools/cavstool.py

```diff
@@ -666,14 +667,24 @@ def win_read(base, start, length):
         log.error("bar4_mmap.size()=%d", bar4_mmap.size())
         raise ie
 
-def win_hdr(base):
-    return struct.unpack("<IIII", win_read(base, 0, 16))
+def winstream_reg_hdr(base):
+    hdr = Regs(bar4_mem + base)
+    hdr.WLEN  = 0x00
+    hdr.START = 0x04
+    hdr.END   = 0x08
+    hdr.SEQ   = 0x0c
```

Comment:
Style: while the use of the Regs class is clean and obvious, strictly these aren't registers, they're fields of `struct sys_winstream` with symbol names "len", "start", "end" and "seq".  I guess being cross-device/cross-language there's room for argument.  But absent any specific reason I guess I'd prefer that the API be as identical as possible.